crawl-ref/source/dat/des/variable/float.des

```diff
@@ -4901,6 +4901,77 @@ MAP
 .cccccccccc
 ENDMAP
 
+NAME:   dreamdust_black_sun_cult
+TAGS:   transparent no_monster_gen
+ORIENT: float
+DEPTH:  D:6-10, Zot, !Zot:$
+WEIGHT: 3
+COLOUR: bm = red
+TILE:   m = dngn_transparent_wall_red
+FTILE:  -mA? = floor_volcanic
+KFEAT:  ! = .
+MARKER: ! = lua:fog_machine { cloud_type='flame', pow_min = 10, pow_max = 10, \
+                              delay = 10, size = 1, excl_rad = 0 }
+: if you.in_branch("D") then
+KMONS:  ? = sun demon
+KMONS:  - = nothing w:50 / demonspawn
```

Comment:
Comments above re: species monsters also apply to unclassed demonspawn, since those are no longer in circulation. I would recommend moving this vault a bit deeper in Dungeon and simply doing some reversal of roles: have a black sun boss with sun demons and possibly some other fire or necromancy themed threats for the later dungeon. Placement would probably start at `D:12` at the earliest, where the black sun would make a great `8` level spawn. If you wish to place the vault earlier at its existing `D:6-10` range, you could consider simply having some lower-level spectral monsters (fast monsters or just those with decent melee damage that spawn in the `D:6-10` range make good spectral candidates). For combining the themes of Fire and Necromancy, there are potentially a lot of viable choices for mid and later Dungeon, but it will need to be something other than unclassed demonspawn.